runtime/bcutil/ConstantPoolMap.cpp

```diff
@@ -117,6 +117,18 @@ ConstantPoolMap::computeConstantPoolMapAndSizes()
 	UDATA doubleSlotCount = 0;
 	UDATA cfrCPCount = _classFileOracle->getConstantPoolCount();
 
+	/* Mark all patch entries as referenced, forces the entry to be added to ROM constantpool */
+	J9ClassPatchMap *map = _context->patchMap();
+	if (map != NULL) {
+		Trc_BCU_Assert_Equals(map->size, cfrCPCount);
+		for (U_16 cfrCPIndex = 0; cfrCPIndex < cfrCPCount; cfrCPIndex++) {
+			if (map->indexMap[cfrCPIndex] == 1) {
+				_constantPoolEntries[cfrCPIndex].isReferenced = true;
```

Comment:
Why is this forcing the entries to be referenced and included in the CP?  This patch appears to work but I don't understand why it's required.  

Why can't we leave the unreferenced entries out of the ROMClass - no one can access them anyway.  It seems like the patching code should be checking if the index is 0 (or some invalid value) and the skipping the patch.  ie in this code: https://github.com/eclipse/openj9/blob/47d7b92dc7e9aeff7b0ffdebf46eeaa785a5ac2d/runtime/jcl/common/sun_misc_Unsafe.cpp#L147